src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -71,10 +71,14 @@ def _get_default_capability(capabilities):
     '''
     Gets the first capability in the collection that has 'default' status.
     If none have 'default' status, gets the first capability that has 'available' status.
+    If still none are available, just gets the first one, which (by elimination) must
+    be unavailable, which at least ensures that the user does not get a confusing StopIteration
+    error.
```

Comment:
Why not just catch the StopIteration and throw an error like "No edition is available to your subscription"?

Reply:
That would be simpler, but previous versions (2.0.22 and earlier) would successfully return all the editions. They would just be shown as being unavailable.

Reply:
Actually, let me implement this a little bit differently. Will send update in a minute.